charts/downloadTab/CSVGenerator.jsdom.test.ts

```diff
@@ -1,4 +1,6 @@
 #! /usr/bin/env yarn jest
+
+// Todo: remove snapshot testing
```

Comment:
What's wrong with snapshot testing?

Reply:
Haha okay this is a bit of a rant, because when I first saw Snapshot testing on Jest I was intrigued and gave it the benefit of the doubt since it's from FB. But after playing with it for a bit I think it has no place in unit testing nor CI tests and should be kept separate.

Ultimately it just alerts you to whether something changed. Not whether some designed logic broke, whether something changed. And change is not what we want to avoid. We want to avoid logic breaks.

I think Snapshots can alert you to places where you should have had unit tests but didn't—to the times when something changed that you weren't working on and didn't expect to change, but other than that just noise in the repo and in the workflow. I would treat it as a wholly separate and different kind of testing than unit testing. In a way our current static site (https://github.com/owid/owid-static) is a snapshot test showing blob changes. I think more of those kind of snapshot tests, perhaps powered by GitHub Actions, could be very valuable.

Unit tests though, at least IMO, are far superior. They force you to write testable, modular code, and then do the arrange, act, and explicitly documented assertions ("Here's how the design is supposed to work"). They should serve as good documentation. They are also very fast. They don't require committing machine generated code to VCS (this one adds 2K LOC to our VCS).

With the current CSV tests I have no real idea what is going on:

```
test("one year-based variable", done => {
        const csvGenerator = setupCSVGenerator(792, [3512])
        testIfCSVMatchesSnapshot(csvGenerator, done, 500)
    })
```

The only assertion here is that nothing changed. Is the CSV supposed to be sorted? Do we support encoding "? What are the column name formatted? Do we end in a newline? 

I don't know any of these implicit design decisions. If I were to work on that code, I would expect the snapshots to change, so I would do a `jest --updateSnapshot` and all the previous implicit decisions would be wiped out, replaced by mine.

Anyway, I had been very intrigued by snapshots, and it's been fun to explore them, but I think we should ditch em.

Reply:
Yep, these tests were my first -- and last -- tango with snapshot testing. Overall, they're a shortcut for catching _any_  regression on a major refactor, especially if there's currently zero units tests on the component being refactored. Like any shortcut, they should be used sparingly.